packages/runtime/src/next/components/tests/controls/checkbox-control.test.tsx

```diff
@@ -0,0 +1,126 @@
+/** @jest-environment jsdom */
+
+import { render, screen } from '@testing-library/react'
+import '@testing-library/jest-dom'
+
+import { ElementData } from '../../../../state/react-page'
+import { randomUUID } from 'crypto'
+import { Page } from '../../page'
+import { act } from 'react-dom/test-utils'
+import { ReactRuntimeProvider } from '../../../context/react-runtime'
+import { ReactRuntime } from '../../../../react'
+import { forwardRef } from 'react'
+import {
+  createMakeswiftPageSnapshot,
+  createRootComponent,
+} from '../../../../utils/tests/element-data-test-test'
+import {
+  Checkbox,
+  CheckboxControlData,
+  CheckboxControlDataTypeKey,
+  CheckboxControlDataTypeValueV1,
+  CheckboxControlDataV0,
+  CheckboxControlDataV1,
+} from '@makeswift/controls'
+
+describe('Page', () => {
+  test('can render CheckboxControl v0 data', async () => {
+    // Arrange
+    const checkboxControlData: CheckboxControlDataV0 = true
+    const TestComponentType = 'TestComponent'
+    const testId = 'test-id'
+    const elementData: ElementData = createRootComponent([
+      {
+        key: randomUUID(),
+        type: TestComponentType,
+        props: {
+          checkbox: checkboxControlData,
+        },
+      },
+    ])
+    const snapshot = createMakeswiftPageSnapshot(elementData)
+    const runtime = new ReactRuntime()
+
+    // Act
+    runtime.registerComponent(
+      forwardRef<HTMLDivElement, { checkbox?: CheckboxControlData }>(({ checkbox }, ref) => {
+        return (
+          <div ref={ref} data-testid={testId}>
+            {checkbox ? 'alpha' : 'beta'}
+          </div>
+        )
+      }),
+      {
+        type: TestComponentType,
+        label: 'TestComponent',
+        props: {
+          checkbox: Checkbox(),
+        },
+      },
+    )
+
+    // Assert
+    const renderResult = await act(async () =>
+      render(
+        <ReactRuntimeProvider runtime={runtime}>
+          <Page snapshot={snapshot} />
+        </ReactRuntimeProvider>,
+      ),
+    )
+
+    expect(renderResult).toMatchSnapshot()
```

Comment:
What I was thinking is that we snapshot the `snapshot` here, so we have a record of the element tree that can be examined visually: `expect(snapshot).toMatchSnapshot()`

Reply:
Got it, that makes more much less noise. Will address.